### PR TITLE
1377 buildx support in ci pipelines

### LINKED ISSUE #1377: Buildx support in CI pipelines
# Summary

We would like to have docker [buildx](https://docs.docker.com/buildx/working-with-buildx) in the CI step, which would help us build multi-platform images with Devtron (which can run within `amd`, `arm` etc processors).

# Motivation

Presently, Devtron runs CI pipelines with `docker build` command which we think is hardcoded as part of CI step. We would want to use `docker buildx` command which supports a lot of features on top of the good old `docker build`. `buildx` would help us in building multi-platform images which can run on all types of processors.
https://docs.docker.com/buildx/working-with-buildx/#build-multi-platform-images

# Proposal

We would like Devtron to provide `buildx` option in the CI pipelines. Few additional options can also be provided to the user to choose which platform images the user wants to build. Or, it can also be provided as part of `Global Configurations`, where the Admins can select the platform(s) (`amd`, `arm` etc) for which all the apps need to build their images.

### PR DESCRIPTION
**Pull request by kripanshdevtron**
# Description

Support Buildx added where user now can add target platform while building docker image

Fixes #1377 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] Verify BuildX Feature for any platform mentioned in drop-down & Run the output image on supported and not supported platform/architecture
- [x] Verify BuildX For all 3 mentioned platform in drop-down & Run the output image on supported and not supported platform/architecture
- [x] Verify BuildX for non-existing platform/architecture and observe logs
- [x] Verify Normal Build feature without setting target platforms(BuildX)
- [x] Verify CPU and Memory for BuildX/Build and observe the difference

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have tested it for all user roles
* [ ] I have added all the required unit/api test cases

### CODE REVIEW

**Comment by deepak-devtron** [comment]
Test Cases:1. Verify BuildX command

File: scripts/sql/61_alter_ci_template.up.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE ci_template
+    ADD COLUMN target_platform VARCHAR(1000) DEFAULT '';
```
Comment (manish-agrawal-ai) [inline scripts/sql/61_alter_ci_template.up.sql:2]: If this column is not nullable then add NOT NULL constraint to it. 
Default value will insert only this value if not provided. 

not adding NOT NULL constraint to this column will result in no error for below query : 
insert into ci_template(target_platform) values (NULL).

File: scripts/sql/61_alter_ci_template.down.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE ci_template
```
Comment (manish-agrawal-ai) [inline scripts/sql/61_alter_ci_template.down.sql:1]: check file number once you merge into main.

Comment (kripanshdevtron) [reply scripts/sql/61_alter_ci_template.down.sql:1]: Sure

File: api/restHandler/CoreAppRestHandler.go [hunk 3]
```diff
@@ -497,6 +497,7 @@ func (handler CoreAppRestHandlerImpl) buildDockerConfig(appId int) (*appBean.Doc
 		BuildConfig: &appBean.DockerBuildConfig{
 			Args:                   ciConfig.DockerBuildConfig.Args,
 			DockerfileRelativePath: ciConfig.DockerBuildConfig.DockerfilePath,
+			TargetPlatform:         ciConfig.DockerBuildConfig.TargetPlatform,
```
Comment (manish-agrawal-ai) [inline api/restHandler/CoreAppRestHandler.go:500]: Please update spec specs/app_create_api.yaml also

**Review by manish-agrawal-ai — APPROVED** [review]

**Pull request merged**